metadata/snippet/params.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2017 The kubecfg authors
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package snippet
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/google/go-jsonnet/ast"
+	"github.com/google/go-jsonnet/parser"
+)
+
+const (
+	componentsID = "components"
+)
+
+func visitComponentsObj(component, snippet string) (*ast.Node, error) {
+	tokens, err := parser.Lex(component, snippet)
+	if err != nil {
+		return nil, err
+	}
+
+	root, err := parser.Parse(tokens)
+	if err != nil {
+		return nil, err
+	}
+
+	switch n := root.(type) {
+	case *ast.Object:
```

Comment:
Ok, so, just to be clear: we aren't allowing imports at the top of the file for now. Right? If not we need to file a bug to come back and fix it. (It should be only a few lines of code, so we could also do it now.)

Reply:
Adding this to a ticket to improve overall handling of complex jsonnet schemas #45